### PR TITLE
[MAINTENANCE] Add unit and integration tests for Splitting on Mod Integer 

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Implement Unit and Integration testing for the "split_on_mod_integer" splitter method on SQLite and All Supported Major SQL Backends.
* Fix implementation issues (proper casting to `Integer` type is required for performing the `mod` operation).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-727/GREAT-902
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py [hunk 1]
```diff
@@ -693,11 +695,11 @@ def get_split_query_for_data_for_batch_identifiers_for_split_on_mod_integer(
         column_name: str,
         mod: int,
     ) -> Selectable:
-        """Divide the values in the named column by `divisor`, and split on that"""
-        # query = f"SELECT DISTINCT(\"{self.column_name}\" / {divisor}) AS my_var FROM {self.table_name}"
+        """Divide the values in the named column by `mod`, and split on that"""
+        # query = f"SELECT DISTINCT(\"{self.column_name}\" % {mod}) AS my_var FROM {self.table_name}"
```
Comment (cdkini) [inline great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py]: Can this be deleted? Is there a reason it is commented out?

**Review by cdkini — APPROVED** [review]

File: great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py]: seconded

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM as well. just one question that @cdkini brought up

File: great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py [hunk 2]
```diff
@@ -228,9 +228,11 @@ def split_on_mod_integer(
         mod: int,
         batch_identifiers: dict,
     ) -> bool:
-        """Divide the values in the named column by `divisor`, and split on that"""
-
-        return sa.column(column_name) % mod == batch_identifiers[column_name]
+        """Divide the values in the named column by `mod`, and split on that"""
```
Comment (anthonyburdi) [inline great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py:279]: Good catch!

File: great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py (hunk 1 above)
Comment (anthonyburdi) [reply great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py]: I don't see why not, a lot of these methods were moved as-is from the execution engine and can use some clean up.

Comment (alexsherstinsky) [reply great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py]: @cdkini I believe that this and similar comments were added by @anthonyburdi  -- I fixed the typos in a couple of them, but am in no position to delete them.  Thanks!

File: tests/integration/test_script_runner.py [hunk 3]
```diff
@@ -655,6 +655,104 @@ class IntegrationTestFixture:
         ),
         extra_backend_dependencies=BackendDependencies.AWS,
     ),
+    IntegrationTestFixture(
```
Comment (anthonyburdi) [inline tests/integration/test_script_runner.py:658]: Maybe we should generate these test fixtures in a for loop?

**Review by anthonyburdi — APPROVED** [review]
LGTM - just one suggestion.

File: tests/integration/test_script_runner.py (hunk 3 above)
Comment (alexsherstinsky) [reply tests/integration/test_script_runner.py:658]: @anthonyburdi They are not all the same...  I need to first reorganize them into "dockerizable" and "non-dockerizable" (all at once, after all the coverage related PRs are merged).  Thanks.

Comment (alexsherstinsky) [reply tests/integration/test_script_runner.py:658]: @anthonyburdi If you are indeed OK with me removing those comments @cdkini pointed out -- how about in a next PR (as this one is far along with Azure checks)?  Thanks!

File: great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py]: Done all that (had to make some fixes to the present PR).

File: tests/integration/test_script_runner.py (hunk 3 above)
Comment (alexsherstinsky) [reply tests/integration/test_script_runner.py:658]: Done.

**Pull request merged**